### PR TITLE
Bufr msg test

### LINKED ISSUE #25: Implement bufr handeling in esoh/ingest/main
The module for handeling bufr is now in a working state. We have to implement passing bufr files to the correct module in the main part of this package.
We also need to implement some test to check that the bufr module supply valid mqtt messages.

### PR DESCRIPTION
**Pull request by istvans-mn**
Closes EURODEO/e-soh-event-queue#25  EURODEO/e-soh#8  BUFR msg handling and tests.  EURODEO/e-soh-event-queue#25  EURODEO/e-soh#8

### CODE REVIEW

**Comment by istvans-mn** [comment]
I added a real [GTS data examples](https://github.com/EURODEO/e-soh-event-queue/pull/42/commits/7dc6c9ace8cf8a1fc1b7af2b94bbf438d134e551) . The pytest is running slower time than usual, because it generates (and validates) 292185 MQTT messages.

File: src/esoh/ingest/main.py [hunk 1]
```diff
@@ -68,10 +74,13 @@ def _decide_input_type(self, message) -> str:
         """
         Internal method for deciding what type of input is being provided.
         """
-        match message.split(".")[-1]:
+        file_name = os.path.basename(message)
+        if re.match("data[0-9][0-9][0-9][0-9]", file_name):
```
Comment (Teddy-1000) [inline src/esoh/ingest/main.py]: Why are this regex expression needed? What file ending is this handeling?

File: test/test_verify_json_payloads.py [hunk 2]
```diff
@@ -8,6 +8,37 @@
...
+    with open("src/esoh/schemas/e-soh-message-spec.json", "r") as file:
+        e_soh_mqtt_message_schema = json.load(file)
+
+    init_bufrtables_py("")
+    init_oscar_py("./src/esoh/ingest/bufr/oscar/oscar_stations_all.json")
+    init_bufr_schema_py("./src/esoh/schemas/bufr_to_e_soh_message.json")
+    msg_build = ingest_to_pipeline(None, None, "testing", testing=True)
+
+    json_payloads = msg_build._build_messages(bufr_file_path, input_type="bufr")
+    destroy_bufrtables_py()
```
Comment (Teddy-1000) [inline test/test_verify_json_payloads.py]: Is it necessary to manually call this manually, I think this should be handled automatically. C++ destructors should be called when an object becomes unreachable. Alternatively we have to use context managers, ref [this thread](https://stackoverflow.com/questions/55452762/pybind11-destructor-not-invoked)

File: src/esoh/ingest/main.py (hunk 1 above)
Comment (istvans-mn) [reply src/esoh/ingest/main.py]: For example filename "data0100" (without extension). This is real norcom GTS data.

File: test/test_verify_json_payloads.py (hunk 2 above)
Comment (istvans-mn) [reply test/test_verify_json_payloads.py]: Yes, in that case the memory is not freed by the constructor.
I replace the pointer with a smart pointer and it automatically deletes.

Comment (istvans-mn) [reply test/test_verify_json_payloads.py]: I changed the static table objects , therefore I removed destroy_bufrtables_py function.

File: src/esoh/ingest/main.py (hunk 1 above)
Comment (Teddy-1000) [reply src/esoh/ingest/main.py]: Ok, I think we should also verify that there is no file ending, incase we get some data with such a name. Is this for all GTS data, or just norcom?
```suggestion
        if re.match(r"data[0-9][0-9][0-9][0-9]$", file_name):
```
Here I have added regex to require end of string. This is to ensure to not capture any other files by mistake.
When using regex it is norm to make the regex expression a raw string, ensuring that nothing is interpreted before the expression is passed to regex. This is done by putting an r in front of the string.

File: test/test_verify_json_payloads.py (hunk 2 above)
Comment (Teddy-1000) [reply test/test_verify_json_payloads.py]: Great, will resolve this then.

File: src/esoh/ingest/main.py (hunk 1 above)
Comment (istvans-mn) [reply src/esoh/ingest/main.py]: These files is created by norcom. I can't access to GTS directly. The norcomm puts the BUFR messages(with GTS header) to these files from GTS. The decoder can handle GTS data directly.

**Review by Teddy-1000 — APPROVED** [review]
Changes reviewed and approved

**Pull request merged**